quickstart.js

```diff
@@ -66,48 +40,91 @@ async function getAccessToken() {
+    },
+    validateStatus: function (status) {
+      return raiseError ? status === 200 : true;
+    }
   });
+  
+  const url = `${baseUrl}/${resourceUrl}`
+  const response = await api.get(url, { params: queryParams });
+  return response;
+}
+
+// Get a participant access token for the specified participant
+// Used for MyDataHelps Embeddables ONLY
+async function getParticipantAccessToken(serviceAccessToken, participantID, scopes) {
+
+  const payload = {
+      "scope": scopes,
+      "grant_type": "delegated_participant",
+      "participant_id": participantID,
+      "client_id": "MyDataHelps.DelegatedParticipant",
+      "client_secret": "secret",
+      "token": serviceAccessToken,
+  }
+
+  const response = await axios.post(tokenUrl, querystring.stringify(payload));
+  const data = response.data;
+  return data.access_token;
 }
 
+// Use this utility if you ever need to print a response for debugging
 function logResponse(response) {
   console.log(util.inspect(response, { colors: true, depth: 3 }));
 }
 
-getAccessToken().then( token => {
-  if (token) {
-    console.log('Obtained access token:');
-    console.log(token);
-
-    const participantResourceUrl = `/api/v1/administration/projects/` + rksProjectId + '/participants';
-    getFromApi(token, participantResourceUrl)
-    .then(data => {
-      if (data) {
-        // Uncomment this to print out the full response to the console.
-        // logResponse(data);
-        console.log(`\nTotal Participants: ${data.totalParticipants}`);
-      } else {
-        console.log("Error when accessing the API.");
-      }
-    });
-  }
-  else {
-    console.log("Error obtaining access token.");
+async function quickstart() {
+  var url;
+  var response;
+  
+  // Get a service access token, needed for all API calls.
+  const serviceAccessToken = await getServiceAccessToken();
+  console.log('Obtained service access token:');
+  console.log(serviceAccessToken);
+
+  // Get all participants
+  url = `/api/v1/administration/projects/${rksProjectId}/participants`;
+  response = await getFromApi(serviceAccessToken, url);
+  const participants = response.data;
+  console.log(`\nTotal Participants: ${participants.totalParticipants}`);
+
+  // Get a specific participant by identifier. We disable 'raiseError' here
+  // so we can handle the 404 case ourselves.
+  const participantIdentifier = "YOUR_PARTICIPANT_IDENTIFIER"
+  
+  if (participantIdentifier != "YOUR_PARTICIPANT_IDENTIFIER") {
+    url = `/api/v1/administration/projects/${rksProjectId}/participants/${participantIdentifier}`;
+    response = await getFromApi(serviceAccessToken, url, {}, false );
+    if (response.status === 404) {
+      console.log("\nParticipant not found.");
+    } else {
+      const participant = response.data;
+      console.log(`\nParticipant Found. ID: ${participant.id}`);
+
+      // NOTE: This piece is only necessary when using MyDataHelps Embeddables in a custom app. 

```

Comment:
Should this comment get moved up to where we isolate the ppt identifier? Just so people who aren't using MDH Embeddable stop sooner rather than later? 

Doesn't matter too much to me.

Reply:
Well I was thinking that illustrating how to query for a particular participant is useful whether you're using embeddables or not. It's only the query for the ppt access token that is embeddables-specific.

Reply:
Agreed with the wider use case of querying ppts